common/src/hash/cpp/Base64.cpp

```diff
@@ -136,7 +136,10 @@ int DecodeBase64(
   size_t buf = 0, len = 0;
 
   while (in < end) {
-    unsigned char c = kDecode[*in++];
+    // In base64 characters are guaranteed to be ASCII, and so are positive
+    // even if char is a signed type.
```

Comment:
Consider runtime check, and terminate if the condition fails. It is a public function, so `in` may be an arbitrary byte array out of our control, not Base64 sequence. In fact, there is no guarantee in caller code, see HashUtils.kt.